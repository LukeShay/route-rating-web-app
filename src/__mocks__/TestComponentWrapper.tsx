/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IViewContextState } from "../context/view/viewStore";
import { IUserContextState } from "../context/user/userStore";
import { StaticRouter } from "react-router";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperProps {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  location: string;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapper: React.FC<ITestComponentWrapperProps> = ({
  children,
  gymsState,
  location,
  userState,
  viewState
}): JSX.Element => {
  return (
    <StaticRouter location={`https://lukeshay.com/${location}`}>
      <GymsStoreMock state={gymsState} dispatch={() => {}}>
        <ViewStoreMock state={viewState} dispatch={() => {}}>
          <UserStoreMock state={userState} dispatch={() => {}}>
            {children}
          </UserStoreMock>
        </ViewStoreMock>
      </GymsStoreMock>
    </StaticRouter>
  );
};

export default TestComponentWrapper;
