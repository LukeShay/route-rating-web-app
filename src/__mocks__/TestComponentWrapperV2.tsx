/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IUserContextState } from "../context/user/userStore";
import { IViewContextState } from "../context/view/viewStore";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperV2Props {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapperV2: React.FC<ITestComponentWrapperV2Props> = ({
  children,
  gymsState,
  userState,
  viewState
}): JSX.Element => {
  return (
    <GymsStoreMock state={gymsState} dispatch={() => {}}>
      <ViewStoreMock state={viewState} dispatch={() => {}}>
        <UserStoreMock state={userState} dispatch={() => {}}>
          {children}
        </UserStoreMock>
      </ViewStoreMock>
    </GymsStoreMock>
  );
};

export default TestComponentWrapperV2;
