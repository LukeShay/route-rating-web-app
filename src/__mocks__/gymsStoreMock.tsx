import { GymsContext, IGymsContextState } from '../context/gyms/gymsStore';
import React from 'react';

export interface IGymsStoreMockProps {
  state: IGymsContextState;
  dispatch: any;
}

const GymsStoreMock: React.FC<IGymsStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <GymsContext.Provider value={{ state, dispatch }}>
    {children}
  </GymsContext.Provider>
);

export default GymsStoreMock;
