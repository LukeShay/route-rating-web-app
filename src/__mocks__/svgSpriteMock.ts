export default {
  id: 'svg-sprite-id',
  viewBox: '0 0 1 1',
  url: '/sprite.svg#svg-sprite-id',
};
