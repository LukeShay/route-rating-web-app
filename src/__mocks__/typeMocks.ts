import { Gym, GymPageType, Route, User, Wall, WallTypes } from '../types';
import { GymsContextState } from '../context/gyms/gymsStore';
import { ViewContextState } from '../context/view/viewStore';
import { UserContextState } from '../context/user/userStore';

export const testRouteOne: Route = {
  averageGrade: 'GRADE_5_9',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Blue',
  id: 'route1',
  name: 'RouteOne',
  setter: 'LukeOne',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testRouteTwo: Route = {
  averageGrade: 'GRADE_5_12ab',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Red',
  id: 'route2',
  name: 'RouteTwo',
  setter: 'LukeTwo',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testWallOne: Wall = {
  gymId: 'gym1',
  id: 'wall1',
  name: 'Wall1',
  routes: [testRouteOne, testRouteTwo],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testWallTwo: Wall = {
  gymId: 'gym1',
  id: 'wall2',
  name: 'Wall2',
  routes: [],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testGymOne: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym1',
  logoUrl: '',
  name: 'Gym One',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testGymTwo: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym2',
  logoUrl: '',
  name: 'Gym Two',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testUserEditor: User = {
  authority: 'ADMIN',
  city: 'Ames',
  country: 'USA',
  email: 'email',
  firstName: 'Name',
  id: 'id',
  lastName: 'Last',
  password: 'password',
  phoneNumber: '1111111111',
  role: 'ADMIN_ROLE',
  session: null,
  state: 'IA',
  username: 'username',
};

export const mockGymsState: GymsContextState = {
  page: { content: [testGymOne, testGymTwo] } as GymPageType,
};

export const darkDesktopState: ViewContextState = {
  mobile: false,
  theme: 'DARK_THEME',
};

export const signedInEditorState: UserContextState = {
  user: testUserEditor,
};

export const signedOutState: UserContextState = {
  user: null,
};
