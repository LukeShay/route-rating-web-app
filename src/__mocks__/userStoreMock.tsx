import { IUserContextState, UserContext } from '../context/user/userStore';
import React from 'react';

export interface IUserStoreMockProps {
  state: IUserContextState;
  dispatch: any;
}

const UserStoreMock: React.FC<IUserStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <UserContext.Provider value={{ state, dispatch }}>
    {children}
  </UserContext.Provider>
);

export default UserStoreMock;
