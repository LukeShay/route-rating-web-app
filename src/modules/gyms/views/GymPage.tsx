import { createStyles, makeStyles, Theme } from "@material-ui/core";
import Button from "@material-ui/core/Button";
import AddIcon from "@material-ui/icons/Add";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";
import RateReviewIcon from "@material-ui/icons/RateReview";
import React from "react";
import { useHistory } from "react-router-dom";
import { toast } from "react-toastify";
import * as GymsActions from "../../../context/gyms/gymsActions";
import { useGymsContext } from "../../../context/gyms/gymsStore";
import { useUserContext } from "../../../context/user/userStore";
import { Routes } from "../../../routes";
import { Gym, Route, Wall } from "../../../types";
import * as StyleUtils from "../../../utils/styleUtils";
import * as GymUtils from "../../../utils/gymUtils";
import GymInformation from "./components/GymInformation";
import RatingAddModal from "./components/RatingAddModal";
import RatingPage from "./RatingPage";
import RouteAddModal from "./components/RouteAddModal";
import RouteEditModal from "./components/RouteEditModal";
import RoutesList from "./components/RoutesList";
import WallAddModal from "./components/WallAddModal";
import WallEditModal from "./components/WallEditModal";
import WallList from "./components/WallList";
import * as ResponseUtils from "../../../utils/responseUtils";
import { useViewContext } from "../../../context/view/viewStore";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    addButton: {
      position: "absolute",
      right: "10px"
    },
    buttonWrapper: {
      marginBottom: theme.spacing(1),
      marginTop: theme.spacing(1)
    },
    icons: {
      paddingRight: theme.spacing(1)
    },
    list: {
      marginTop: theme.spacing(2),
      paddingBottom: "20px"
    }
  })
);

const GymPage: React.FC = (): JSX.Element => {
  const { state: gymsState, dispatch: gymsDispatch } = useGymsContext();
  const { state: userState } = useUserContext();
  const { state: viewState } = useViewContext();

  const [gym, setGym] = React.useState<Gym>({} as Gym);
  const [view, setView] = React.useState<"WALL" | "ROUTE" | "RATING">("WALL");
  const [wall, setWall] = React.useState<Wall | undefined>(undefined);
  const [routes, setRoutes] = React.useState<Route[]>([]);
  const [route, setRoute] = React.useState<Route | undefined>(undefined);
  const [canEdit, setCanEdit] = React.useState<boolean>(false);
  const [wallId, setWallId] = React.useState<string>("");
  const [openAdd, setOpenAdd] = React.useState<boolean>(false);
  const [openEdit, setOpenEdit] = React.useState<boolean>(false);

  const classes = useStyles();
  const history = useHistory();
  const [gymId] = React.useState<string | undefined>(
    history.location.pathname
      .split("/")
      .splice(-1)
      .pop()
  );

  React.useEffect(() => {
    if (!gymId || !gymsState.page || !gymsState.page.content) {
      history.push(Routes.GYMS);
    } else {
      const tempGym = GymUtils.getGymById(gymsState.page.content, gymId);

      if (!tempGym) {
        history.push(Routes.GYMS);
      } else if (!tempGym.walls) {
        if (tempGym.id) {
          GymsActions.loadWalls(gymsDispatch, tempGym).then(
            (response: Response) => {
              ResponseUtils.toastIfNotOk(response, "Error getting walls.");
            }
          );
        }
      }

      if (tempGym) {
        setGym(tempGym);

        const { user } = userState;
        const { authorizedEditors } = tempGym;

        const tempWall = GymUtils.getWallById(tempGym, wallId);

        if (tempWall) {
          setRoutes(tempWall.routes);
          setWallId(tempWall.id);

          if (route) {
            const tempRoute = tempWall.routes
              ? tempWall.routes.find(
                  (element: Route) => route.id === element.id
                )
              : null;

            if (tempRoute) {
              setRoute(tempRoute);
            }
          }
        }

        if (
          user &&
          authorizedEditors &&
          authorizedEditors.find((editorId: string) => editorId === user.id)
        ) {
          setCanEdit(true);
        } else {
          setCanEdit(false);
        }
      }
    }
  }, [gymId, gymsState, userState]);

  const handleWallRowClick = async (rowWallId: string): Promise<void> => {
    const tempWall = GymUtils.getWallById(gym, rowWallId);

    if (tempWall) {
      if (!tempWall.routes || tempWall.routes.length === 0) {
        const response = await GymsActions.loadRoutes(
          gymsDispatch,
          gym,
          tempWall.id
        );

        if (!ResponseUtils.isOk(response)) {
          toast.error("Error getting routes.");
        }
      }

      setView("ROUTE");
      setRoutes(tempWall.routes);
      setWallId(tempWall.id);
    } else {
      toast.error("Could not find wall.");
    }
  };

  const handleDeleteWall = async (rowWallId: string): Promise<void> => {
    if (
      window.confirm(
        "Are you sure you want to delete this wall? This action cannot be undone."
      )
    ) {
      if (gymId) {
        GymsActions.deleteWall(gymsDispatch, rowWallId, gym).then(
          (response: Response) => {
            if (!response || !(response instanceof Response) || !response.ok) {
              toast.error("Error deleting wall.");
            }
          }
        );
      } else {
        toast.error("Error deleting wall.");
      }
    }
  };

  const handleRouteRowClick = async (routeFromRow: Route): Promise<void> => {
    setRoute(routeFromRow);
    setView("RATING");
  };

  const handleDeleteRoute = async (routeId: string): Promise<void> => {
    if (
      window.confirm(
        "Are you sure you want to delete this route? This action cannot be undone."
      )
    ) {
      if (gymId) {
        GymsActions.deleteRoute(
          gymsDispatch,
          { id: routeId, gymId, wallId } as Route,
          gym
        ).then((response: Response) => {
          ResponseUtils.toastIfNotOk(response, "Error deleting route.");
        });
      } else {
        toast.error("Error deleting route.");
      }
    }
  };

  const handleEditRoute = async (tempRoute: Route): Promise<void> => {
    if (canEdit) {
      setRoute(tempRoute);
      setOpenEdit(true);
      setOpenAdd(false);
    }
  };

  const handleEditWall = async (tempWall: Wall): Promise<void> => {
    if (canEdit) {
      setWall(tempWall);
      setOpenEdit(true);
      setOpenAdd(false);
    }
  };

  const handleOpenAdd = async (): Promise<void> => {
    if ((view !== "RATING" && canEdit) || view === "RATING") {
      setOpenAdd(true);
      setOpenEdit(false);
    }
  };

  const handleCloseAdd = async (): Promise<void> => setOpenAdd(false);

  const handleCloseEdit = async (): Promise<void> => {
    setOpenEdit(false);
    setRoute(undefined);
    setWall(undefined);
  };

  const Buttons: React.FC = (): JSX.Element => (
    <div className={classes.buttonWrapper}>
      <Button
        variant="text"
        fullWidth={false}
        size="medium"
        type="button"
        onClick={(): void => {
          if (view === "ROUTE") {
            setView("WALL");
            setWallId("");
          } else {
            setView("ROUTE");
            setRoute(undefined);
          }
        }}
        style={StyleUtils.shouldBeVisible(view !== "WALL")}
      >
        <ArrowBackIcon className={classes.icons} />
        Back
      </Button>
      <Button
        onClick={handleOpenAdd}
        className={classes.addButton}
        variant="text"
        fullWidth={false}
        size="medium"
        type="button"
        style={StyleUtils.shouldBeVisible(
          canEdit || (view === "RATING" && userState.user !== null)
        )}
      >
        {view === "RATING" ? (
          <RateReviewIcon className={classes.icons} />
        ) : (
          <AddIcon className={classes.icons} />
        )}
        {view === "RATING" ? "Review" : "Add"}
      </Button>
    </div>
  );

  const CurrentView: React.FC = (): JSX.Element => {
    if (view === "WALL") {
      return (
        <WallList
          walls={gym.walls}
          onRowClick={handleWallRowClick}
          canEdit={canEdit}
          handleDeleteWall={handleDeleteWall}
          handleEditClick={handleEditWall}
        />
      );
    }

    if (view === "ROUTE") {
      return (
        <RoutesList
          canEdit={canEdit}
          routes={routes}
          handleDeleteRoute={handleDeleteRoute}
          handleEditRoute={handleEditRoute}
          handleRowClick={handleRouteRowClick}
        />
      );
    }

    if (view === "RATING") {
      return <RatingPage route={route || ({} as Route)} />;
    }

    return <React.Fragment />;
  };

  const Modals: React.FC = (): JSX.Element => {
    if (gymId) {
      return (
        <React.Fragment>
          <RouteAddModal
            open={view === "ROUTE" && openAdd}
            handleClose={handleCloseAdd}
            gym={gym}
            wallId={wallId}
          />
          {route && (
            <RouteEditModal
              open={view === "ROUTE" && openEdit}
              handleClose={handleCloseEdit}
              gym={gym}
              wallId={wallId}
              route={route}
            />
          )}
          <WallAddModal
            open={view === "WALL" && openAdd}
            handleClose={handleCloseAdd}
            gym={gym}
          />
          {wall && (
            <WallEditModal
              open={view === "WALL" && openEdit}
              handleClose={handleCloseEdit}
              gymId={gymId}
              wall={wall}
            />
          )}
          {route && (
            <RatingAddModal
              open={view === "RATING" && openAdd}
              handleClose={handleCloseAdd}
              routeId={route.id}
              gym={gym}
              wallId={wallId}
            />
          )}
        </React.Fragment>
      );
    } else {
      return <React.Fragment />;
    }
  };

  return (
    <React.Fragment>
      <GymInformation gym={gym} canEdit={canEdit} mobile={viewState.mobile} />
      <div
        className={classes.list}
        style={StyleUtils.shouldDisplay(
          (gym.walls && gym.walls.length !== 0) || canEdit
        )}
      >
        <Buttons />
        <CurrentView />
        <Modals />
      </div>
    </React.Fragment>
  );
};

export default React.memo(GymPage);
