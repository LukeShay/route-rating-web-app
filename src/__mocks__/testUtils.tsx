import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import TestComponentWrapperV2 from './TestComponentWrapperV2';
import * as TypeMocks from './typeMocks';

interface WrapperProps {
  children: React.ReactNode;
}

const SignedInDesktop: React.FC<WrapperProps> = ({ children }): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedInEditorState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedInDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedInDesktop, ...options });

const SignedOutDesktop: React.FC<WrapperProps> = ({
  children,
}): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedOutState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedOutDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedOutDesktop, ...options });

export * from '@testing-library/react';
