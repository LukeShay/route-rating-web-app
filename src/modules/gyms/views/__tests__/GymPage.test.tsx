import React from "react";
import * as TypeMocks from "../../../../__mocks__/typeMocks";
import { shallowToJson } from "enzyme-to-json";
import { shallow } from "../../../../../configs/setupEnzyme";
import GymPage from "../GymPage";
import TestComponentWrapper from "../../../../__mocks__/TestComponentWrapper";

let wrapper;

describe("<GymPage />", () => {
  beforeEach(() => {
    wrapper = shallow(
      <TestComponentWrapper
        gymsState={TypeMocks.mockGymsState}
        location={`/gyms/${TypeMocks.testGymOne.id}`}
        userState={TypeMocks.signedInEditorState}
        viewState={TypeMocks.darkDesktopState}
      >
        <GymPage />
      </TestComponentWrapper>
    );
  });

  it("should render correctly when editor.", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
