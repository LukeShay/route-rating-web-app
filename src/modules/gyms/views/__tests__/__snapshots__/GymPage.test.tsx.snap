// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GymPage /> should render correctly when editor. 1`] = `
<StaticRouter
  location="https://lukeshay.com//gyms/gym1"
>
  <GymsStoreMock
    dispatch={[Function]}
    state={
      Object {
        "page": Object {
          "content": Array [
            Object {
              "address": "1234 Five Street",
              "authorizedEditors": Array [
                "id",
              ],
              "city": "Ames",
              "email": "abc@d.com",
              "id": "gym1",
              "logoUrl": "",
              "name": "Gym One",
              "phoneNumber": "1234567890",
              "photoUrl": "",
              "state": "Iowa",
              "walls": Array [
                Object {
                  "gymId": "gym1",
                  "id": "wall1",
                  "name": "Wall1",
                  "routes": Array [
                    Object {
                      "averageGrade": "GRADE_5_9",
                      "averageRating": 4,
                      "gymId": "gym1",
                      "holdColor": "Blue",
                      "id": "route1",
                      "name": "RouteOne",
                      "setter": "Luke",
                      "types": Array [
                        "TOP_ROPE",
                        "LEAD",
                      ],
                      "wallId": "wall1",
                    },
                    Object {
                      "averageGrade": "GRADE_5_12ab",
                      "averageRating": 4,
                      "gymId": "gym1",
                      "holdColor": "Blue",
                      "id": "route2",
                      "name": "RouteTwo",
                      "setter": "Luke",
                      "types": Array [
                        "TOP_ROPE",
                        "LEAD",
                      ],
                      "wallId": "wall1",
                    },
                  ],
                  "types": Array [
                    "LEAD",
                    "TOP_ROPE",
                  ],
                },
                Object {
                  "gymId": "gym1",
                  "id": "wall2",
                  "name": "Wall1",
                  "routes": Array [],
                  "types": Array [
                    "LEAD",
                    "TOP_ROPE",
                  ],
                },
              ],
              "website": "lukeshay.com",
              "zipCode": "50014",
            },
            Object {
              "address": "1234 Five Street",
              "authorizedEditors": Array [
                "id",
              ],
              "city": "Ames",
              "email": "abc@d.com",
              "id": "gym2",
              "logoUrl": "",
              "name": "Gym Two",
              "phoneNumber": "1234567890",
              "photoUrl": "",
              "state": "Iowa",
              "walls": Array [
                Object {
                  "gymId": "gym1",
                  "id": "wall1",
                  "name": "Wall1",
                  "routes": Array [
                    Object {
                      "averageGrade": "GRADE_5_9",
                      "averageRating": 4,
                      "gymId": "gym1",
                      "holdColor": "Blue",
                      "id": "route1",
                      "name": "RouteOne",
                      "setter": "Luke",
                      "types": Array [
                        "TOP_ROPE",
                        "LEAD",
                      ],
                      "wallId": "wall1",
                    },
                    Object {
                      "averageGrade": "GRADE_5_12ab",
                      "averageRating": 4,
                      "gymId": "gym1",
                      "holdColor": "Blue",
                      "id": "route2",
                      "name": "RouteTwo",
                      "setter": "Luke",
                      "types": Array [
                        "TOP_ROPE",
                        "LEAD",
                      ],
                      "wallId": "wall1",
                    },
                  ],
                  "types": Array [
                    "LEAD",
                    "TOP_ROPE",
                  ],
                },
                Object {
                  "gymId": "gym1",
                  "id": "wall2",
                  "name": "Wall1",
                  "routes": Array [],
                  "types": Array [
                    "LEAD",
                    "TOP_ROPE",
                  ],
                },
              ],
              "website": "lukeshay.com",
              "zipCode": "50014",
            },
          ],
        },
      }
    }
  >
    <ViewStoreMock
      dispatch={[Function]}
      state={
        Object {
          "mobile": false,
          "theme": "DARK_THEME",
        }
      }
    >
      <UserStoreMock
        dispatch={[Function]}
        state={
          Object {
            "user": Object {
              "authority": "ADMIN",
              "city": "Ames",
              "country": "USA",
              "email": "email",
              "firstName": "Name",
              "id": "id",
              "lastName": "Last",
              "password": "password",
              "phoneNumber": "1111111111",
              "role": "ADMIN_ROLE",
              "session": null,
              "state": "IA",
              "username": "username",
            },
          }
        }
      >
        <Memo(GymPage) />
      </UserStoreMock>
    </ViewStoreMock>
  </GymsStoreMock>
</StaticRouter>
`;
